tools/ci/check-changelog.js

```diff
@@ -0,0 +1,46 @@
+const fs = require("fs");
+const proc = require("child_process");
+
+const skipChangelogInfix = "[no-changelog]";
+const changelogPath = process.argv[2];
+const baseRef = process.argv[3];
+
+function wasChangelogModified() {
+  const diffArgs = [
+    "--no-pager",
+    "diff",
+    "--exit-code",
+    baseRef,
+    "--",
+    changelogPath,
+  ];
+  const result = proc.spawnSync("git", diffArgs);
+  const exitCode = result.status;
+  console.log(result.stdout.toString("utf-8"));
+  const noDifference = exitCode == 0;
+  return !noDifference;
```

Comment:
What happens here if you get an `ENOENT` for git?

Reply:
Oh, good question, I haven't thought about it. Well in theory the GH runners should have git, but better safe than sorry.

I checked and the `status` is then `null`. But in the meantime I do `stdout.toString` and as `stdout` is also `null`, it crashes with `Cannot read property 'toString' of null`.

But I guess an explicit check would be better, so I will add it.